tests/conftest.py

```diff
@@ -787,13 +812,19 @@ class FakePackage:
 
     name: str
     version: str
-    filename: str
+    source_file: Union[str, Path]
     metadata: MetadataKind
     # This will override any dependencies specified in the actual dist's METADATA.
     requires_dist: Tuple[str, ...] = ()
     # This will override the Name specified in the actual dist's METADATA.
     metadata_name: Optional[str] = None
 
+    @property
+    def filename(self) -> str:
+        if isinstance(self.source_file, str):
+            return self.source_file
```

Comment:
Should this do a `os.path.basename`? (Probably not very important either way)

Reply:
Oh yes, that interface was supposed to be temporary! Let me fix that.

Reply:
Done! Added a new `FakePackageSource` dataclass to handle this.